app/services/nodes/merger.rb

```diff
@@ -90,15 +92,33 @@ def update_properties
   end
 
   def copy_attachments
-    self.copied_attachments = []
+    self.copied_attachments = {}
 
     source_node.attachments.each do |attachment|
-      copied_attachments << attachment.copy_to(target_node)
+      new_attachment = attachment.copy_to(target_node)
+      copied_attachments[attachment.filename] = new_attachment
+    end
+    update_attachment_references
+  end
+
+  def update_attachment_references
+    target_node.evidence_ids.each do |evidence_id|
+      evidence = Evidence.find(evidence_id)
+      evidence.content.scan(Attachment::SCREENSHOT_REGEX).each do |screenshot_path|
```

Comment:
This is very similar to the `AttachmentsCopier#copy_attachments` method. Is there something we can reuse from there?